mmdet3d/ops/sst_modules/sst_ops.py

```diff
@@ -0,0 +1,200 @@
+            - The number of token of each window in this batching.
+
+    Returns:
+        Tensor: The feat in sequence format. has shape (N, FEAR_DIM)
+    """
+
+    num_voxels = sum([len(item[0]) for item in seq_win_mapping_list])
+
+    feat_dim = region_batching_feat_list[0].size(-1)
+    seq_feat = region_batching_feat_list[0].new_zeros((num_voxels, feat_dim))
+    for region_batching_feat, seq_win_mapping in zip(region_batching_feat_list,
+                                                     seq_win_mapping_list):
+        index_in_all_windows, voxel_inds, num_tokens_each_win = seq_win_mapping
+        region_batching_feat = region_batching_feat.reshape(-1, feat_dim)
+        single_seq_feat = region_batching_feat[index_in_all_windows]
+        seq_feat[voxel_inds] = single_seq_feat
+    return seq_feat
+
+
+def seq_to_win(feat, seq_win_mapping_list):
+    """Transform the feat from a sequence (N, D) to several batching windows
+    (B, NUM_TOKENS, D).
+
+    Args:
+        feat (tensor): The sequence features to be transformed.
+        seq_win_mapping_list (list[tuple]): Each tuple contains
+            three important information of corresponding
+            batching.
+
+            - A identical index of voxel in corresponding
+              batching windows
+            - The index of voxel in original sequence.
+            - The number of token of each window in this batching.
+
+    Returns:
+        list(tensor): List of region batching feats, each tensor
+        has shape (NUM_WINS, NUM_TOKEN, FEAT_DIM).
+    """
+
+    feat_dim = feat.shape[-1]
+    region_batching_feat_list = []
+
+    for (index_in_all_windows, voxel_inds,
+         num_tokens_each_win) in seq_win_mapping_list:
+        feat_in_batching = feat[voxel_inds]
+        num_windows = (index_in_all_windows //
+                       num_tokens_each_win).max().item() + 1
+        region_batching_feat = feat.new_zeros(
+            (num_windows * num_tokens_each_win, feat_dim))
+        region_batching_feat[index_in_all_windows] = feat_in_batching
+        region_batching_feat = region_batching_feat.reshape(
+            (num_windows, num_tokens_each_win, feat_dim))
+        region_batching_feat_list.append(region_batching_feat)
+
+    return region_batching_feat_list
+
+
+def get_inner_win_inds(win_inds):
+    """Calculate the index of each voxel inner windows.
+
+    The code is modified from original implementation
+    https://github.com/TuSimple/SST
+
+    Args:
+        win_inds (Tensor): The value indicates which windows a voxel
+            belongs to. Voxels share a window have same inds.
+            has shape (N, ).
+
+    Return:
+        ori_inner_inds (Tensor): The value indicates voxel's id
+        in a window. if M voxels share a window, their
+        inner_inds would be torch.arange(m, dtype=torch.long)
+    """
+
+    # the logic may be complicated
+    # We add an example here, win_inds = [0, 1, 2, 0, 0, 1]
```

Comment:
Thanks for your conscientious work !!!
It seems that the example should be `win_inds = [0, 2, 2, 0, 0, 1]` or `win_inds = [0, 1, 2, 0, 0, 2]` according to following comments. There should be two `2` .

Reply:
Thanks for your wonderful work, I will fix it.